daemon/restart.go

```diff
@@ -34,11 +35,23 @@ func (daemon *Daemon) ContainerRestart(name string, seconds *int) error {
 // gracefully stop, before forcefully terminating the container. If
 // given a negative duration, wait forever for a graceful stop.
 func (daemon *Daemon) containerRestart(container *container.Container, seconds int) error {
+
+	// Determine isolation. If not specified in the hostconfig, use daemon default.
```

Comment:
Yes, in that scenario, it would switch. Without that behaviour, it could potentially mean a container export from server and into client (at least pre RS5 where client could only run Hyper-V containers), or visa-versa, you would have a container that couldn't start. But yes, entirely orthogonal.